src/ORM/DB.php

```diff
@@ -109,6 +126,124 @@ public static function get_conn($name = 'default')
         return null;
     }
 
+    /**
+     * Only use the primary database connection for the rest of the current request
+     *
+     * @internal
+     */
+    private static bool $mustUsePrimary = false;
+
+    /**
+     * Used by DB::withPrimary() to count the number of times it has been called
+     * Uses an int instead of a bool to allow for nested calls     *
+     *
+     * @internal
+     */
+    private static int $withPrimaryCount = 0;
+
+    /**
+     * The key of the previously choosen random replica config
+     *
+     * @internal
+     */
+    private static string $randomReplicaConfigKey = '';
```

Comment:
```suggestion
     * The key of the replica config to use for this request
     *
     * @internal
     */
    private static string $readOnlyReplicaConfigKey = '';
```
It doesn't matter that it's selected randomly - what matters is what it represents, which is the key for the read-only replica that will be used for the whole request.

Reply:
Updated